db/crud.go

```diff
@@ -1464,7 +1464,7 @@ Truth table for AllowConflicts and noConflicts combinations:
                        AllowConflicts=true     AllowConflicts=false
    noConflicts=true    continue checks         continue checks
    noConflicts=false   return false            continue checks */
-func (db *Database) IsIllegalConflict(doc *Document, parentRevID string, deleted, noConflicts bool) bool {
+func (db *Database) IsIllegalConflict(doc *Document, parentRevID string, deleted, noConflicts bool, newFlag bool) bool {
```

Comment:
newFlag needs a better name, but as implemented I think this represents 'is this an active sgreplicate peer'.  

Aside from changing the name to be more descriptive, the concern I have here is that this won't apply to passive, Hydrogen sg-replicate peers.  I think we need it to apply in that case as well, to allow tombstones of non-leaf nodes to be replicated.

Reply:
Woops my bad yeah. I intended to rename that. Thanks for catching it. Okay. I can add that handling